src/server/tiered_storage.cc

```diff
@@ -63,6 +63,16 @@ void RecordAdded(const PrimeValue& pv, size_t tiered_len, DbTableStats* stats) {
   stats->tiered_used_bytes += tiered_len;
 }
 
+string DecodeString(bool is_raw, string_view str, PrimeValue decoder) {
+  if (is_raw) {
+    decoder.Materialize(str, true);
+    string tmp;
+    decoder.GetString(&tmp);
+    return tmp;
+  }
+  return string{str};
+}
```

Comment:
why do you need to decoder, can't you mateiralize an empty compact objcet?

Reply:
answered above